src/main/java/com/strandls/cca/service/impl/CCADataServiceImpl.java

```diff
@@ -688,4 +708,92 @@ public Boolean sendPermissionGrant(HttpServletRequest request, EncryptedKey encr
 		return false;
 	}
 
+
+	@Override
+	public Map<String, Object> searchCCAData(String query, HttpServletRequest request, UriInfo uriInfo)
+	        throws JsonProcessingException {
+	    try {
+	        MultivaluedMap<String, String> queryParams = uriInfo.getQueryParameters();
+
+	        // Set default values for offset and limit parameters if not provided
+	        int offset = Integer.parseInt(queryParams.getOrDefault("offset", Collections.singletonList("0")).get(0));
+	        int limit = Integer.parseInt(queryParams.getOrDefault("limit", Collections.singletonList("100")).get(0));
+
```

Comment:
please verify if we need to hard code a value 100 here

Reply:
I have used these values as default in case the user/ui doesn't assign them while making the API calls.
Should I remove them or place these constants in some file like CCAConstants.java?